src/pages/experiments/[experimentId]/data-exploration/components/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -46,92 +47,133 @@ const DiffExprCompute = (props) => {
   useEffect(() => {
     if (hierarchy.length === 0) return;
 
-    const newSelectableClusters = _.cloneDeep(hierarchy);
-    // create a new item for each hierarchy to represent All
-    // eslint-disable-next-line array-callback-return
-    newSelectableClusters.map(({ key, children }) => {
-      if (children && children.length > 0) {
-        children.unshift(generateSpecialKey(key));
+    setSelectedGroups(_.mapValues(selectedGroups, (cellSetKey) => {
+      if (!cellSetKey) {
+        return null;
       }
-    });
-    setSelectableClusters(newSelectableClusters);
-    setSelectedCellSets(_.mapValues(selectedCellSets, (cellSetKey) => {
-      if (isKeySpecial(cellSetKey)) {
-        return 'All';
+
+      const splitKey = cellSetKey.split('/')[1];
+
+      if (cellSetKey === 'all' || cellSetKey === 'background' || splitKey === 'rest') {
+        return cellSetKey;
       }
-      if (cellSetKey !== defaultSelected && !properties[cellSetKey]) {
-        return defaultSelected;
+
+      if (!properties[splitKey]) {
+        return null;
       }
+
       return cellSetKey;
     }));
   }, [hierarchy, properties]);
 
   const validateForm = () => {
-    if (selectedCellSets.cellSet === defaultSelected) {
+    if (!selectedGroups.cellSet) {
       setIsFormValid(false);
       return;
     }
-    if (selectedCellSets.compareWith === defaultSelected) {
+
+    if (!selectedGroups.compareWith) {
       setIsFormValid(false);
       return;
     }
-    if (selectedCellSets.cellSet === selectedCellSets.compareWith) {
+
+    if (!selectedGroups.basis) {
       setIsFormValid(false);
       return;
     }
+
     setIsFormValid(true);
   };
 
+  // Validate form when the groups selected changes.
   useEffect(() => {
     validateForm();
-  }, [selectedCellSets]);
+  }, [selectedGroups]);
 
   /**
    * Updates the selected clusters.
    * @param {string} cellSet The key of the cell set.
    * @param {string} option The option string (`cellSet` or `compareWith`).
    */
   const onSelectCluster = (cellSet, option) => {
-    setSelectedCellSets({
-      ...selectedCellSets,
-      [option]: cellSet,
+
+    setSelectedGroups({
+      ...selectedGroups,
+      [option]:
+        cellSet,
```

Comment:
specifying `option` here as a key list creates the need for having more complicated logic in all places where we will be handling these selected groups. Is there any reason why we don't instead structure this like this:

```
...
selectOption: [option],
cellset, 
...
```
instead of 
```
[option]:
   cellset,
```

Reply:
`option` refers to `cellSet`, `compareWith`, and `basis`. The created data structure is essentially:

```
{
    cellSet: 'louvain-0',
    compareWith: 'background',
    basis: 'all'
}
```

which gets passed straight into `DifferentialExpression` as a work body. I don't see how the proposal would work with this data structure. Maybe you misunderstood what the `option` parameter is, or do you think refactoring is still the best idea? Would be happy to discuss.

Reply:
we discussed over zoom and it is good actually, I misunderstood it